cmd/dp-dataset-exporter/main.go

```diff
@@ -85,22 +89,69 @@ func main() {
 
 	eventHandler := event.NewExportHandler(filterStore, observationStore, fileStore, eventProducer)
 
-	go event.Consume(kafkaConsumer, eventHandler, errorHandler)
+	eventConsumer := event.NewConsumer()
+	eventConsumer.Consume(kafkaConsumer, eventHandler, errorHandler)
 
-	<-signals
+	shutdownGracefully := func() {
 
-	close(exit)
+		ctx, cancel := context.WithTimeout(context.Background(), config.GracefulShutdownTimeout)
 
-	// gracefully dispose resources
-	kafkaConsumer.Closer() <- true
-	kafkaProducer.Closer() <- true
+		// gracefully dispose resources
+		err = eventConsumer.Close(ctx)
+		logIfError(err)
 
-	err = dbConnection.Close()
-	if err != nil {
-		log.Error(err, log.Data{"message": "failed to close connection to Neo4j"})
+		err = kafkaConsumer.Close(ctx)
+		logIfError(err)
+
+		err = kafkaProducer.Close(ctx)
```

Comment:
Should we not close our producers after shutting down server.

For instance, if a kafka message is to be written due to an ongoing request then it wont be written in the current ordering of closing connections to either servers or kafka.

Reply:
Any existing requests should be complete by the time eventConsumer.Close(ctx) returns, so there should be no issues with the order of closing Kafka resources.